mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearches.java

```diff
@@ -621,37 +623,60 @@ private boolean after(Date date1, Date date2, DateResolution res) {
     }
 
     private String createDateString(Date date, DateResolution res) {
-        SimpleDateFormat format;
+        SimpleDateFormat format = createFormat(res);
+        format.setCalendar(getGMT());
+        return format.format(date);
+    }
+
+    private SimpleDateFormat createFormat(DateResolution res) {
         switch (res) {
         case Year:
-            format = new SimpleDateFormat("yyyy");
-            break;
+            return new SimpleDateFormat("yyyy");
         case Month:
-            format = new SimpleDateFormat("yyyyMM");
-            break;
+            return new SimpleDateFormat("yyyyMM");
         case Day:
-            format = new SimpleDateFormat("yyyyMMdd");
-            break;
+            return new SimpleDateFormat("yyyyMMdd");
```

Comment:
Hum this is probably out of the scope of the PR but I think if this is this way we implement filter on day this is bad. Day depend of the timezone of the user, what do jmap spec specify about that by the way